### PR TITLE
Do not allow retroactive revocation for DIP-150

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======
Need to Remove retroactive revocation and clarify how to handle reversing undesired historical actions.

Link to GitHub Issue(s): DIP-#150

Solution
========
Made validation of a Batch immutable

Change summary:
---------------
- [x] Updated Spec Versions
- [x] Added several non-normative sections clarifying how to handle situations
- [x] PR for the contracts repo with the interface changes: https://github.com/LibertyDSNP/contracts/pull/49

### CODE REVIEW

File: pages/DSNP/BatchPublications.md [hunk 1]
```diff
@@ -71,3 +76,10 @@ Applications need to know if a given Batch file has any information they are int
...
 1. JSON, BSON, and SQLite, while used for storage sometimes, are intended for serialization. They are schemaless, which results in redundant information and therefore a larger size than formats with schemas. They also don't support Bloom filters; instead, indexing would be required, or new batches would need to be downloaded entirely.  The exception is SQLite, which does support more advanced queries; however, it was designed for in-memory storage.
+
+### Batch Correctness and Order
+
+Batch correctness is immutable and usually in part based on the validation of the delegation of authors listed inside the batch to the publisher.
+Due to the nature of distributed systems, it is possible that a race condition occurs such that a user's delegation revocation presents before a Batch that contains a message from that user via the revoked delegate.
+While those individual messages should be considered invalid, allowing amount of time to pass before considering the entire batch invalid is suggested.
```
Comment (saraswatpuneet) [inline pages/DSNP/BatchPublications.md:84]: We can say acceptance window ?

File: pages/DSNP/Identity.md [hunk 2]
```diff
@@ -24,3 +24,17 @@ the ability to delegate permission to others to announce content on the user's b
...
+
+
+## Non-normative
+
+### Retroactive Revocation of Delegation
+
+There are many times when someone would desire retroactive revocation of delegation.
+If a key was found to have been compromised at an earlier time for example.
+However, retroactive revocation is much more difficult from a caching and performance perspective.
+Instead (reverting any undesirable Announcements)(Announcements.md#reverting-an-announcement) such as through [Tombstones](Types/Tombstone.md) allows a user to choose the specific events that need reverting and notify the network of that change.
```
Comment (saraswatpuneet) [inline pages/DSNP/Identity.md:37]: Did you want to make it a link? Missing square braces

Comment (saraswatpuneet) [reply pages/DSNP/Identity.md:37]: Around "reverting any undesirable announcements"

File: pages/DSNP/Identity.md [hunk 3]
```diff
@@ -24,3 +24,17 @@ the ability to delegate permission to others to announce content on the user's b
...
+
+To this end, any [Batch Publications](BatchPublications.md) can always be validated from the perspective of the time they were published instead of at read time.
+The validity of a Batch is thus immutable.
```
Comment (saraswatpuneet) [inline pages/DSNP/Identity.md:40]: I like this approach how will require some discussion around how to track such edge cases to know delegation was removed after the published time but not important for this PR

**Review by saraswatpuneet — APPROVED** [review]
Left few comments 🎉

File: pages/DSNP/Identity.md (hunk 3 above)
Comment (wilwade) [reply pages/DSNP/Identity.md:40]: Tried adding some steps below to make it more clear.

File: pages/DSNP/BatchPublications.md (hunk 1 above)
Comment (wilwade) [reply pages/DSNP/BatchPublications.md:84]: I like that. I think (wrt your other comment on Identity.md) we can make it a leading window only as well! I'll try something out and let me know.

**Review by enddynayn — APPROVED** [review]

**Review by shannonwells — APPROVED** [review]

**Pull request merged**